packages/mobile/scripts/run_app.sh

```diff
@@ -0,0 +1,123 @@
+# -f: Fast (skip steps not required unless network or depedencies changes)
+# -h: Hot Reload (Restore nav state on reload)
+
+NETWORK=""
+PLATFORM=""
+FAST=false
+HOT_RELOAD=false
+while getopts 'n:p:fh' flag; do
+  case "${flag}" in
+    n) NETWORK="$OPTARG" ;;
+    p) PLATFORM="$OPTARG" ;;
+    f) FAST=true ;;
+    h) HOT_RELOAD=true ;;
+    *) error "Unexpected option ${flag}" ;;
+  esac
+done
+
+[ -z "$PLATFORM" ] && echo "Need to set the PLATFORM via the -p flag" && exit 1;
+
+# Get machine type (needed later)
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     MACHINE=Linux;;
+    Darwin*)    MACHINE=Mac;;
+    CYGWIN*)    MACHINE=Cygwin;;
+    MINGW*)     MACHINE=MinGw;;
+    *)          MACHINE="UNKNOWN:${unameOut}"
+esac
+echo "Machine type: $MACHINE"
+echo "Current directory: `pwd`"
+
+# Read values from the .env file and put them in env vars
+ENV_FILENAME="${ENVFILE:-.env}"
+export $(grep -v '^#' $ENV_FILENAME | xargs)
+
+if [ -z "$NETWORK" ]; then
+  echo "No network set."
+  read -p "Use $DEFAULT_TESTNET network set in .env file (y/n)? "
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    NETWORK=$DEFAULT_TESTNET
+  else 
+    echo "No network chosen. Exiting."
+    exit 1
+  fi
+fi
+
+# Set DEFAULT_TESTNET in .env file
+sed -i.bak "s/DEFAULT_TESTNET=.*/DEFAULT_TESTNET=$NETWORK/g" $ENV_FILENAME
+
+# Set Hot Reload (saved nav state) in .env file
+sed -i.bak "s/DEV_RESTORE_NAV_STATE_ON_RELOAD=.*/DEV_RESTORE_NAV_STATE_ON_RELOAD=$HOT_RELOAD/g" $ENV_FILENAME
+
+# Set Firebase settings in google service config files
+ANDROID_GSERVICES_PATH="./android/app/src/debug/google-services.json"
+IOS_GSERVICES_PATH="./ios/GoogleService-Info.plist"
+sed -i.bak "s/celo-org-mobile-.*firebaseio.com/celo-org-mobile-$NETWORK.firebaseio.com/g" $ANDROID_GSERVICES_PATH
+sed -i.bak "s/celo-org-mobile-.*firebaseio.com/celo-org-mobile-$NETWORK.firebaseio.com/g" $IOS_GSERVICES_PATH
+
+# Cleanup artifacts from in-place sed replacement on BSD based systems (macOS)
+rm -f $ENV_FILENAME.bak
+rm -f $ANDROID_GSERVICES_PATH.bak
+rm -f $IOS_GSERVICES_PATH.bak
+
+
+# Build Wallet Kit for env
+if [ "$FAST" = false ]; then
+  echo "Building sdk for testnet $NETWORK"
+  yarn build:sdk $NETWORK
+  echo "Done building sdk"
+fi
+
+# Build the app and run it
+if [ $PLATFORM = "android" ]; then
+  echo "Using platform android"
+
+  NUM_DEVICES=`adb devices -l | wc -l`
+  if [ $NUM_DEVICES -lt 3 ]; then 
+    echo "No android devices found"
+    exit 1
+  fi
+
+  # Run jettify to fix non-android-x compatible libs
+  if [ "$FAST" = false ]; then
+    echo "Jetifying react native libraries"
+    cd ../../ && yarn run jetify && cd packages/mobile
+    echo "Jetified"
+  fi
+
+  if [ $MACHINE = "Mac" ]; then
+    echo "Starting packager in new terminal"
+    RN_START_CMD="cd `pwd`;yarn react-native start"
+    OSASCRIPT_CMD="tell application \"Terminal\" to do script \"$RN_START_CMD\""
+    echo "FULLCMD: $OSASCRIPT_CMD"
+    osascript -e "$OSASCRIPT_CMD"
```

Comment:
Would be cool to do that only if the packager is not already running.
I usually start the packager separately so I can launch it with custom options like `--reset-cache`.

Reply:
What does reset cache do? Would you like  a flag  for the script  that sets it?

Reply:
It removes cached files, makes sure all files are transformed again. Fixes edge cases when changing babel config or transformer config.
Not sure it's worth having a flag for now here.